nullaway/src/main/java/com/uber/nullaway/Nullness.java

```diff
@@ -210,10 +212,34 @@ public static boolean hasNullableAnnotation(Symbol symbol, Config config) {
    */
   public static boolean paramHasNullableAnnotation(
       Symbol.MethodSymbol symbol, int paramInd, Config config) {
+    // We treat the (generated) equals() method of record types to have a @Nullable parameter, as
+    // the generated implementation handles null (as required by the contract of Object.equals())
+    if (isRecordEqualsParam(symbol, paramInd)) {
+      return true;
+    }
     return hasNullableAnnotation(
         NullabilityUtil.getAllAnnotationsForParameter(symbol, paramInd), config);
   }
 
+  private static boolean isRecordEqualsParam(Symbol.MethodSymbol symbol, int paramInd) {
+    if (!symbol.owner.getKind().toString().equals("RECORD")) {
```

Comment:
We need `toString()` here (vs an enum equality check of some sort) for older JDK compat, right? Maybe worth a quick comment if so?